python/python/bystro/ancestry/listener.py

```diff
@@ -38,30 +106,99 @@ def _load_queue_conf(queue_conf_path: str) -> QueueConf:
     return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
 
 
+def _fill_missing_data(genotypes: pd.DataFrame) -> tuple[pd.DataFrame, pd.Series]:
+    sample_missingnesses = genotypes.isna().mean(axis="columns")  # average over columns, leaving indices
+    # todo: much better imputation strategy to come, but we're stubbing out for now.
+    # if col completely missing in all samples, just fill as heterozygote for now
```

Comment:
Is the result exactly the same if all samples are all homozygous ref or all hets?

Reply:
If there's at least one sample for which genotyping data for a given variant is present, we'll take the mean dosage over all samples where we have data.  If it's completely absent from the dataset, however, we're arbitrarily filling it in as heterozygote just to get a complete input matrix.

This isn't a great imputation strategy-- in fact, it's about the worst one.  I'm preparing another PR to do imputation in a more sophisticated way-- this is just a placeholder to get the model running end to end and will be replaced as soon as we can demonstrate that capability.

Meanwhile lmk if there's any way I can make that clearer in the code?